src/main/java/duke/SaveManager.java

```diff
@@ -0,0 +1,173 @@
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import duke.exception.DukeSaveDataException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+
+/** Contains functions for saving into and loading from save file */
+public class SaveManager {
+
+    /** Represents save file directory */
+    private Path saveFilePath;
+
+    /**
+     * Creates a new <code>SaveManager</code>.
+     *
+     * @param saveFilePath save file directory.
+     */
+    public SaveManager(Path saveFilePath) {
+        this.saveFilePath = saveFilePath;
+    }
+
+    /**
+     * Loads save data from save file at <code>saveFilePath</code>.
+     * If unable to load, returns an empty <code>TaskManager</code> and attempts to create a new save data file.
+     *
+     * @return <code>TaskManager</code> containing tasks loaded from save file.
+     * @throws DukeSaveDataException Unable to load save data.
+     */
+    public TaskManager load() throws DukeSaveDataException {
+
+        try {
+            // Initialise variables
+            TaskManager taskManager = new TaskManager();
+
+            // Loop for reading line by line (i.e. task by task) from save file to taskManager
+            BufferedReader br = Files.newBufferedReader(this.saveFilePath);
+            String currLine = br.readLine();
+            while (currLine != null) {
+                taskManager.storeTask(this.loadTaskFromSave(currLine));
+                currLine = br.readLine();
+            }
+
+            return taskManager;
+
+        } catch (NoSuchFileException e) {
+            // If unable to load file, attempts to create the file directory and a new, blank save file
+            try {
+                Files.createDirectories(this.saveFilePath.getParent());
+                Files.createFile(this.saveFilePath);
+
+            } catch (IOException e2) {
+                // Throw DukeSaveDataException if unable to create a new save file
+                throw new DukeSaveDataException("Unable to load from save file or create new save file");
+            }
+
+            // Throw DukeSaveDataException for not being able to load from save file
+            throw new DukeSaveDataException("Unable to load from save file - " + e.toString());
+        } catch (IOException e) {
+            throw new DukeSaveDataException("Unable to load from save file - " + e.toString());
+        }
+
+    }
+
+    /**
+     * Saves data from given <code>TaskManager</code> to save file in <code>saveFilePath</code>.
+     *
+     * @param taskManager Contains task data to save.
+     * @throws DukeSaveDataException If unable to save data.
+     */
+    public void save(TaskManager taskManager) throws DukeSaveDataException {
+
+        // Initialise save data storage variable
+        // Currently implemented as a String[] for use with taskManager.forEach()
+        final String[] saveData = {""};
```

Comment:
Similar to previous comment, could perhaps rename it to plural form for better code readability.

Reply:
While this is implemented as a list, it should not be used as a list i.e. only the first slot is used at any time. Basically this is not used as a Collection. This is similar in functionality to a class that just serves as a wrapper i.e.
```
class Temp {
    public String saveData;
}
```
or else `TaskManager.forEach()` will not be able to modify the `saveData`. But that breaks a bunch of rules too, and it seems too elaborate to properly create getters and setters for a temp class. What do you think?